openmetadata-ui/src/main/resources/ui/src/components/common/TestConnection/TestConnection.tsx

```diff
@@ -0,0 +1,340 @@
+        intervalId = toNumber(
+          setInterval(async () => {
+            const workflowResponse = await getWorkflowData(response.id);
+            const { response: testConnectionResponse } = workflowResponse;
+
+            const isWorkflowFailed =
+              workflowResponse.status === WorkflowStatus.Failed;
+            const isWorkflowSuccess =
+              workflowResponse.status === WorkflowStatus.Successful;
+
+            const isWorkflowCompleted = isWorkflowFailed || isWorkflowSuccess;
+
+            const isTestConnectionSuccess =
+              (testConnectionResponse?.status as StatusType) ===
+              StatusType.Successful;
+
+            const isTestConnectionFailed =
+              (testConnectionResponse?.status as StatusType) ===
+              StatusType.Failed;
+
+            if (isWorkflowCompleted) {
+              setMessage(
+                isTestConnectionSuccess
+                  ? t('message.connection-test-successful')
+                  : t('message.connection-test-failed')
+              );
+
+              isTestConnectionSuccess &&
+                showSuccessToast(t('message.connection-test-successful'));
+
+              isTestConnectionFailed &&
+                showErrorToast(t('message.connection-test-failed'));
+
+              // clear the current interval
+              intervalId && clearInterval(intervalId);
+
+              // set testing connection to false
+              setIsTestingConnection(false);
+
+              // set the test connection status
+              setTestStatus(
+                isTestConnectionSuccess
+                  ? StatusType.Successful
+                  : StatusType.Failed
+              );
+            }
+          }, FETCH_INTERVAL)
+        );
+
+        // stop fetching the workflow after 2 minutes
+        setTimeout(() => {
+          // clear the current interval
+          intervalId && clearInterval(intervalId);
+
+          // using reference to ensure call back should have latest value
+          const currentWorkflowStatus = currentWorkflowRef.current
+            ?.status as WorkflowStatus;
+
+          const isWorkflowCompleted = WORKFLOW_COMPLETE_STATUS.includes(
+            currentWorkflowStatus
+          );
+
+          if (!isWorkflowCompleted) {
+            const infoMessage = t('message.test-connection-taking-too-long');
+            setMessage(infoMessage);
+            showInfoToast(infoMessage);
+          }
+
+          setIsTestingConnection(false);
+        }, FETCHING_EXPIRY_TIME);
+      }
+    } catch (error) {
+      intervalId && clearInterval(intervalId);
+      showErrorToast(error as AxiosError);
+      setIsTestingConnection(false);
+      setMessage(t('message.connection-test-failed'));
+      setTestStatus(StatusType.Failed);
+    }
+  };
+
+  useEffect(() => {
+    currentWorkflowRef.current = currentWorkflow; // update ref with latest value of currentWorkflow state variable
+  }, [currentWorkflow]);
```

Comment:
Not sure why we want to update ref with state, Ref should does the Job correctly